src/components/PageSettingsModal.jsx

```diff
@@ -0,0 +1,248 @@
+        data: params,
+        withCredentials: true,
+      });
+
+      window.location.reload();
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  changeHandler = (event) => {
+    const { id, value } = event.target;
+    let errorMessage = '';
+    switch (id) {
+      case 'permalink': {
+        // Permalink is too short
+        if (value.length < PERMALINK_MIN_LENGTH) {
+          errorMessage = `The permalink should be longer than ${PERMALINK_MIN_LENGTH} characters.`;
+        }
+
+        // Permalink is too long
+        if (value.length > PERMALINK_MAX_LENGTH) {
+          errorMessage = `The permalink should be shorter than ${PERMALINK_MAX_LENGTH} characters.`;
+        }
+
+        // Permalink fails regex
+        if (!permalinkRegexTest.test(value)) {
+          console.log('IN REGEX FAIL', value);
+          errorMessage = `The permalink should start and end with slashes and contain 
+            lowercase words separated by hyphens only.
+            `;
+        }
+        break;
+      }
+      case 'title': {
+        // Title is too short
+        if (value.length < TITLE_MIN_LENGTH) {
+          errorMessage = `The title should be longer than ${TITLE_MIN_LENGTH} characters.`;
+        }
+        // Title is too long
+        if (value.length > TITLE_MAX_LENGTH) {
+          errorMessage = `The title should be shorter than ${TITLE_MAX_LENGTH} characters.`;
+        }
+        break;
+      }
+      default:
+        break;
+    }
+    this.setState({
+      errors: {
+        [id]: errorMessage,
+      },
+      [id]: value,
+    });
+  }
+
+  render() {
+    const { title, permalink, errors } = this.state;
+    const { settingsToggle, isNewPage } = this.props;
+
+    // Page settings form has errors - disable save button
+    const hasErrors = _.some(errors, (field) => field.length > 0);
+
+    return (
+      <div className={elementStyles.overlay}>
+        <div className={elementStyles.modal}>
+          <div className={elementStyles.modalHeader}>
+            <h1>{ isNewPage ? 'Create new page' : 'Update page settings'}</h1>
+            <button type="button" onClick={settingsToggle}>
+              <i className="bx bx-x" />
+            </button>
+          </div>
+          <form className={elementStyles.modalContent} onSubmit={this.saveHandler}>
+            <div className={elementStyles.modalFormFields}>
+              <p className={elementStyles.formLabel}>Title</p>
+              <input
+                value={title}
+                id="title"
+                required
+                autoComplete="off"
+                onChange={this.changeHandler}
+                className={errors.title ? `${elementStyles.error}` : null}
+              />
+              <span className={elementStyles.error}>
+                {' '}
+                {errors.title}
+                {' '}
+              </span>
+              <p
```

Comment:
Perhaps we can create a `PageSettingsModalElement` since there is duplication of code between the `Title` and `Permalink` fields

Reply:
I think a better abstraction is to group the following 3 HTML elements:
```
<p className={elementStyles.formLabel}>{label}</p>
<input {...props} />
<span className={elementStyles.error}>{errors[label]}</span>
```

Reply:
Will do it in a following PR

Reply:
Logged in #55